pkg/asset/manifests/powervs/cluster.go

```diff
@@ -0,0 +1,238 @@
+		cosRegion          string
+		imageName          string
+		bucketName         string
+		err                error
+		powerVSCluster     *capibm.IBMPowerVSCluster
+		powerVSImage       *capibm.IBMPowerVSImage
+	)
+
+	defer func() {
+		logrus.Debugf("GenerateClusterAssets: installConfig = %+v, clusterID = %v, bucket = %v, object = %v", installConfig, *clusterID, bucket, object)
+		logrus.Debugf("GenerateClusterAssets: ic.ObjectMeta = %+v", installConfig.Config.ObjectMeta.Name)
+		logrus.Debugf("GenerateClusterAssets: installConfig.Config.PowerVS = %+v", *installConfig.Config.PowerVS)
+		logrus.Debugf("GenerateClusterAssets: resourceGroupName = %v", resourceGroupName)
+		logrus.Debugf("GenerateClusterAssets: vpcName = %v", vpcName)
+		logrus.Debugf("GenerateClusterAssets: vpcRegion = %v", vpcRegion)
+		logrus.Debugf("GenerateClusterAssets: transitGatewayName = %v", transitGatewayName)
+		logrus.Debugf("GenerateClusterAssets: cosName = %v", cosName)
+		logrus.Debugf("GenerateClusterAssets: cosRegion = %v", cosRegion)
+		logrus.Debugf("GenerateClusterAssets: imageName = %v", imageName)
+		logrus.Debugf("GenerateClusterAssets: bucketName = %v", bucketName)
+		logrus.Debugf("GenerateClusterAssets: powerVSCluster.Spec.ControlPlaneEndpoint.Host = %v", powerVSCluster.Spec.ControlPlaneEndpoint.Host)
+	}()
+
+	manifests = []*asset.RuntimeFile{}
+
+	network = fmt.Sprintf("%s-network", clusterID.InfraID)
+
+	if installConfig.Config.PowerVS.ServiceInstanceGUID == "" {
+		serviceName := fmt.Sprintf("%s-iaas", clusterID.InfraID)
+
+		service = capibm.IBMPowerVSResourceReference{
+			Name: &serviceName,
+		}
+	} else {
+		service = capibm.IBMPowerVSResourceReference{
+			ID: &installConfig.Config.PowerVS.ServiceInstanceGUID,
+		}
+	}
+
+	if installConfig.Config.Platform.PowerVS.PowerVSResourceGroup != "" {
+		// @TODO is it a name or id?
+		resourceGroupName = installConfig.Config.Platform.PowerVS.PowerVSResourceGroup
+	}
+
+	vpcName = fmt.Sprintf("vpc-%s", clusterID.InfraID)
+
+	if installConfig.Config.PowerVS.VPCRegion == "" {
+		if vpcRegion, err = powervstypes.VPCRegionForPowerVSRegion(installConfig.Config.PowerVS.Region); err != nil {
+			return nil, fmt.Errorf("unable to derive vpcRegion from region: %s %w", installConfig.Config.PowerVS.Region, err)
+		}
+	} else {
+		vpcRegion = installConfig.Config.Platform.PowerVS.VPCRegion
+	}
+
+	transitGatewayName = fmt.Sprintf("%s-tg", clusterID.InfraID)
+
+	cosName = fmt.Sprintf("%s-cos", clusterID.InfraID)
+
+	switch {
+	case installConfig.Config.Platform.PowerVS.VPCRegion != "":
+		cosRegion, err = powervstypes.COSRegionForVPCRegion(installConfig.Config.Platform.PowerVS.VPCRegion)
+		if err != nil {
+			return nil, fmt.Errorf("GenerateClusterAssets: COSRegionForVPCRegion returns %w", err)
+		}
+	case installConfig.Config.Platform.PowerVS.Region != "":
+		cosRegion = installConfig.Config.Platform.PowerVS.Region
+	default:
+		return nil, fmt.Errorf("GenerateClusterAssets: Region is empty")
+	}
```

Comment:
Is this being used only to print `cosRegion` to the debug logs?

Reply:
Yes, it seems so.  Removed.